projects/cm_mcu/ZynqMonTask.c

```diff
@@ -380,41 +520,9 @@ void zm_set_fpga(struct zynqmon_data_t data[], int start)
 }
 
 #ifdef REV1
-// this code will ultimately be generated from the YAML file
-void zm_fill_structs(void)
-{
-  // firefly, size 25
-  zm_set_firefly_temps(&zynqmon_data[0], 0);
-  // psmon, size 64
-  zm_set_psmon(&zynqmon_data[25], 32);
-  // adcmon, size 21
-  zm_set_adcmon(&zynqmon_data[89], 96);
-  // uptime, size 2
-  zm_set_uptime(&zynqmon_data[110], 192);
-  // gitversion, size 10.0
-  zm_set_gitversion(&zynqmon_data[112], 118);
-  // fpga, size 4
-  zm_set_fpga(&zynqmon_data[122], 128);
-}
 #define ZMON_VALID_ENTRIES 126
 #elif defined REV2
-// this code will is generated from the YAML file
-void zm_fill_structs(void)
-{
-  // firefly, size 20
-  zm_set_firefly_temps(&zynqmon_data[0], 0);
-  // psmon, size 80
-  zm_set_psmon(&zynqmon_data[20], 32);
-  // adcmon, size 21
-  zm_set_adcmon(&zynqmon_data[104], 128);
-  // uptime, size 2
-  zm_set_uptime(&zynqmon_data[125], 192);
-  // gitversion, size 20
-  zm_set_gitversion(&zynqmon_data[127], 118);
-  // fpga, size 8
-  zm_set_fpga(&zynqmon_data[137], 150);
-}
-#define ZMON_VALID_ENTRIES 145
+#define ZMON_VALID_ENTRIES 233
```

Comment:
aren't these now set in `Zynqmon_Addresses.c`?

Reply:
you're right. However, I got a compile error for not declaring them in this file. I think some lines in the same file such as [this](https://github.com/apollo-lhc/cm_mcu/blob/78ae126eeb029e2c4a4f7a6b1d28e2ab8be7bde0/projects/cm_mcu/ZynqMonTask.c#L531) rely on them. 

Should we declare these two variables in Task.h? instead of both ZynMon*.cc

Reply:
Maybe we should instead pass it in as an argument to the `zm_send_data`? That is probably cleaner anyhow.

Reply:
i.e., modify the `mcu_generate.py` to make the function `void zm_fill_structs(int valid_entries)`

Reply:
actually I guess that would not solve the problem.